### ISSUE TITLE
[jasper:x64-windows] build failure in VS2022

### ISSUE THREAD

**Issue by SuperAoao** [microsoft/vcpkg#38192]
### Operating system

Windows 11

### Compiler

MSVC 19.39.33523

### Steps to reproduce the behavior

```Shell
in manifest mode, my vcpkg.json content:
{
  "dependencies": [
    "osg"
  ]
}
```

### Failure logs

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/14974761/install-x64-windows-dbg-out.log)
[vcpkg-manifest-install.log](https://github.com/microsoft/vcpkg/files/14974767/vcpkg-manifest-install.log)

### Additional context

error: building jasper:x64-windows failed with: BUILD_FAILED
Elapsed time to handle jasper:x64-windows: 21 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jasper
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[jasper]+Build+error+on+x64-windows&body=Copy+issue+body+from+D%3A%2FDevGitHub%2Fbuild%2FOSG_playground%2Fout%2Fbuild%2Fx64-debug%2Fvcpkg_installed%2Fvcpkg%2Fissue_body.md
  
 The error message is very similar to the one in https://github.com/microsoft/vcpkg/issues/30850.

**Comment by MonicaLiu0311** [comment]
Sorry I can't repro your issue with MSVC 19.39.33523.

Similar issue: https://github.com/microsoft/vcpkg/issues/36103.
Has been fixed: https://github.com/microsoft/vcpkg/pull/36143.

Please try again with the latest `builtin-baseline`:
```
PS G:\vcpkg> git rev-parse HEAD
ad3bae57455a3c3ce528fcd47d8e8027d0498add
```

**Comment by SuperAoao** [comment]
I manually changed the 'baseline' from '000d1bda1ffa95a73e0b40334fa4103d6f4d3d48' to 'ad3bae57455a3c3ce528fcd47d8e8027d0498add', and it worked! I simply utilized the 'vcpkg new application' command to generate the vcpkg-configuration.json file and I am unsure why it reverted to an older version of the vcpkg baseline. Nonetheless, I sincerely appreciate your assistance! I have also followed you on GitHub.
